ptf/sainat.py

```diff
@@ -0,0 +1,1493 @@
+                self.client, self.ingr_lag_rif, nat_zone_id=1)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_lag_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_svi_rif, nat_zone_id=1)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_svi_rif, nat_zone_id=0)
+
+            print("\n***Egress L3 port <-***")
+            verify_translation(self.egr_port_rif)
+            print("\n***Egress L3 LAG <-***")
+            verify_translation(self.egr_lag_rif)
+            print("\n***Egress SVI <-***")
+            verify_translation(self.egr_svi_rif)
+
+        finally:
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_svi_rif, ingress_acl=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_lag_rif, ingress_acl=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_port_rif, ingress_acl=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_svi_rif, nat_zone_id=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_svi_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_lag_rif, nat_zone_id=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_lag_rif, nat_zone_id=0)
+
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.ingr_port_rif, nat_zone_id=0)
+            sai_thrift_set_router_interface_attribute(
+                self.client, self.egr_port_rif, nat_zone_id=0)
+
+            sai_thrift_remove_nat_entry(self.client, svi_dnat_pool)
+            sai_thrift_remove_nat_entry(self.client, svi_dnat)
+
+            sai_thrift_remove_nat_entry(self.client, lag_dnat_pool)
+            sai_thrift_remove_nat_entry(self.client, lag_dnat)
+
+            sai_thrift_remove_nat_entry(self.client, port_dnat_pool)
+            sai_thrift_remove_nat_entry(self.client, port_dnat)
+
+
+class NatTest(SaiHelper):
+    '''
+    Basic NAT configuration test
+    '''
+
+    def setUp(self):
+        super(NatTest, self).setUp()
+
+        self.rif_list = []
+        self.nhop_list = []
+        self.nbor_list = []
+        self.route_list = []
+        self.nat_list = []
+
+        self.nbor_mac = '00:11:22:33:44:55'
+        self.dmac = '00:22:22:22:22:22'
+        self.src_ip = '192.168.0.10'
+        self.dst_ip = '10.10.10.1'
+        self.nhop_ip = '20.20.20.1'
+        self.server_ip = '192.168.0.1'
+        self.server_dmac = '00:33:44:55:66:77'
+        self.translate_server_ip = '200.200.200.1'
+        mask = '/24'
+
+        # Route configuration
+        server_port = self.port24
+        wan_port = self.port25
+
```

Comment:
In previous class NatTranslationTest, we are testing for Lag and SVi also. Is there a reason we do that and not in this class?